### ISSUE TITLE
🐛 `lint/noBarrelFile` reports files that are not barrel files

### ISSUE THREAD

**Issue by Conaclos** [biomejs/biome#1982]
### Environment information

```block
Playground
```

### What happened?

A file without re-export is reported.

[Playground](https://biomejs.dev/playground/?lintRules=all&files.decl.d.ts=ZQB4AHAAbwByAHQAIABjAG8AbgBzAHQAIABBACAAPQAgADAA)

### Expected result

Files without re-exports should not be reported.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
By the way, we should certainly skip declaration files (`.d.ts`).

**Comment by ematipico** [comment]
@togami2864 To understand what kind of file the analyzer is analysing, you can use this API:

```rs
let file_source = ctx.source_type::<JsFileSource>();
```

**Comment by togami2864** [comment]
Thank you for reporting! This bug is misleading:(
